### ISSUE TITLE
[Linux Mint] Error 05: You don't have 1 ID1 in your Nintendo 3DS folder, you have 0!

### ISSUE THREAD

**Issue by LuineGR** [zoogie/MSET9#42]
Hi there. I'm using Linux Mint 22 "Wilma", Linux kernel version 5.15.0-119-generic. Python3 version is 3.10.12.
Yes, I have UTF8 enabled.
I have an Old 3DS XL with version 1.17.
Python script logs are as follows:

`MSET9 v2.0 SETUP by zoogie, Aven, DannyAAM and thepikachugamer
What is your console model and version?
Old 3DS has two shoulder buttons (L and R)
New 3DS has four shoulder buttons (L, R, ZL, ZR)

-- Please type in a number then hit return --

Enter one of these four numbers!
Enter 1 for: Old 3DS/2DS, 11.8.0 to 11.17.0
Enter 2 for: New 3DS/2DS, 11.8.0 to 11.17.0
Enter 3 for: Old 3DS/2DS, 11.4.0 to 11.7.0
Enter 4 for: New 3DS/2DS, 11.4.0 to 11.7.0
>>> 1
[--] Checking if SD card is writeable...
[OK] SD card is writeable!
[--] Ensuring extracted files exist...
[OK] boot9strap.firm looks good!
[OK] boot.firm looks good!
[OK] boot.3dsx looks good!
[OK] b9 looks good!
[OK] SafeB9S.bin looks good!
[--] Detected ID0: 429ec663ef049e1683557d90c54e9cca
[--] Checking databases...
[OK] title.db looks good!
[OK] import.db looks good!
[--] Checking for HOME Menu extdata...
[--] Checking for Mii Maker extdata...
[--] Found file in ID0 folder? 'b2a0014b47606ccb5344313600275048_user-id1'
[XX] Error 05: You don't have 1 ID1 in your Nintendo 3DS folder, you have 0!
[--] Consult: https://3ds.hacks.guide/troubleshooting#installing-boot9strap-mset9 for help!
[--] Press Enter to exit...
`

I followed all listed steps in [the guide](https://3ds.hacks.guide/installing-boot9strap-(mset9-cli)), but my user data **was _not_** temporarily wiped. The Settings app still recognised that the data was "corrupt" and reset, but everything is exactly the same as it was before (even my Miis were still there as they always are, but for some reason the Mii Maker still created extra data for the SD card).

**Comment by LuineGR** [comment]
Quick addendum: [the troubleshooting guide](https://3ds.hacks.guide/troubleshooting.html) only has instructions to deal with a non-zero number of ID1s, which leads me to believe that my situation is somewhat uncommon. I would've otherwise referred to it.

**Comment by luciasfeliz** [comment]
Hey! I am having the same issue.

_I'm very panicked about it. I offered to homebrew a friend's 2DS. This guide is new to me, I've never had to install script for 3DS homebrew._ :sob:

**Comment by Buggem** [comment]
> Hey! I am having the same issue.

Bad SD, not related to mset9 itself. Probably due to shoving it in and out too much

It's funny how I got downvoted and I was actually right. 👎

**Comment by benjaminlammers** [comment]
I encountered this issue and managed to resolve it. The circumstances that lead to this might not be the same for you but this is what I believe the issue to be.

I attempted to use a 128gb SD card under the assumption it would be supported as FAT32 filesystems can be up to 2tb in size. Turns out, for some odd compatibility reason, a 3DS only accepts SD cards up to 32gb in size. Linux based systems have no trouble formatting a storage device larger than 32gb to FAT32, but the 3DS will be upset silently, preventing the exploit from being set up and working properly. I think the reason this is not a well known issue is because Windows will not allow an SD card with the storage capacity of mine to be formatted to FAT32, only permitting exFAT and NTFS without using diskpart. With most people using Windows to setup the exploit this is probably why it's an undocumented issue.

In conclusion, your SD _probably_ shouldn't exceed 32gb in size. If it does, only format 32gb worth of a partition. Assuming you were doing the same thing as me.

**Comment by AdamJ-Collaborator** [comment]
> I encountered this issue and managed to resolve it. The circumstances that lead to this might not be the same for you but this is what I believe the issue to be.
> 
> I attempted to use a 128gb SD card under the assumption it would be supported as FAT32 filesystems can be up to 2tb in size. Turns out, for some odd compatibility reason, a 3DS only accepts SD cards up to 32gb in size. Linux based systems have no trouble formatting a storage device larger than 32gb to FAT32, but the 3DS will be upset silently, preventing the exploit from being set up and working properly. I think the reason this is not a well known issue is because Windows will not allow an SD card with the storage capacity of mine to be formatted to FAT32, only permitting exFAT and NTFS without using diskpart. With most people using Windows to setup the exploit this is probably why it's an undocumented issue.
> 
> In conclusion, your SD _probably_ shouldn't exceed 32gb in size. If it does, only format 32gb worth of a partition. Assuming you were doing the same thing as me.

Thanks for posting this, you might have helped me out - bought a 64GB card and hitting the same ID1 issue. Will see if formatting half the card resolves it!

**Comment by sclera23** [comment]
I had the same issue like yesterday on Bazzite. I ended up switching to the Windows side of my pc to restart the whole process there and it ended up working. From that I can at least say that the issue wasn't due to the SD card (or cards since I actually tried two identical 16 GB ones). I'm pretty certain that the issue wasn't because of formatting either, since I formatted both SD's to FAT32 in Linux and never actually reformatted them when I went to Windows. My best guess is that there was an issue with Linux recognizing the ID1 folder or something of that nature-- I'm not entirely sure though because I modded another 3DS nearly a month ago using the same method and I didn't run into any issues.
